edgedetector/solver/solver.py

```diff
@@ -1,7 +1,105 @@
+from edgedetector.solver.evaluator.cost.dissimilarity_matrix import DissimilarityMatrix
+from edgedetector.solver.evaluator.cost.overall_cost_evaluator import OverallCostEvaluator
+from edgedetector.solver.evaluator.fitness.fitness_evaluator import FitnessEvaluator
+from edgedetector.solver.mutation.mutation_factory import MutationFactory
+from edgedetector.solver.population.initializer_factory import InitializerFactory
+from edgedetector.solver.selection.selection_factory import SelectionFactory
+from edgedetector.solver.stopcondition.stop_condition_factory import \
+    StopConditionFactory
 
 
 class Solver:
-    pass
+
+    def __init__(self, config):
+        self.config = config
+        image_file = config["data"]["inputPath"]
+        self.output_file = config["data"]["outputPath"]
+        self.image = ImageReader().read(image_file)
+        self.population = self.__initialize_population()
+        self.fitness_evaluator = self.__initialize_fitness_evaluator()
+        self.dissimilarity_matrix = DissimilarityMatrix(self.image.image_matrix).matrix
+        self.cost_evaluator = self.__initialize_overall_cost_evaluator()
+        self.crossover = self.__initialize_crossover()
+        self.mutation = self.__initialize_mutation()
+        self.selection = self.__initialize_selection()
+        self.stop_condition = self.__initialize_stop_condition()
+
+    def solve(self):
+        best_fitness, best_genotype = self.stop_condition.run(self.__generation)
+        self.image.edge_matrix = best_genotype.genes
+        ImageWriter().write(self.image, self.output_file)
+
+    def __generation(self):
+        best_fitness, best_genotype = self.__evaluate()
+        self.__breed(best_fitness, best_genotype)
+        return best_fitness, best_genotype
+
+    def __evaluate(self):
+        worst_cost, worst_genotype = self.__count_costs()
+        fitness = self.__count_fitness(worst_genotype)
+        return fitness
+
+    def __count_costs(self):
+        worst_cost = 0
+        worst_genotype = None
+        for genotype in self.population:
+            cost = self.cost_evaluator.evaluate(genotype)
+            if cost > worst_cost:
+                worst_cost = cost
+                worst_genotype = genotype
+        return worst_cost, worst_genotype
+
+    def __count_fitness(self, worst_genotype):
+        best_fitness = 0
+        best_genotype = None
+        self.fitness_evaluator.worst_genotype = worst_genotype
+        for genotype in self.population:
+            fitness = self.fitness_evaluator.evaluate(genotype)
+            if fitness > best_fitness:
+                best_fitness = fitness
+                best_genotype = genotype
+        return best_fitness, best_genotype
+
+    def __breed(self, best_fitness, best_genotype):
+        # while not whole population is new # todo what condition ?
+        #   self.selection.select(...)      # todo not implemented yet
```

Comment:
Leave this one as is. I would like to suggest the following flow:
1. Merge of this pull request.
2. Merge of my pull request containing implementation of the selection mechanism.
3. Then, I will change this line with invocation of a method returning two mates (genotypes) selected for crossover.